src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Text.cs

```diff
@@ -99,33 +101,40 @@ public override bool CheckTypes(TexlNode[] args, DType[] argTypes, IErrorContain
                 return isValid;
             }
 
-            StrLitNode formatNode;
             if (!DType.String.Accepts(argTypes[1]))
             {
                 errors.EnsureError(DocumentErrorSeverity.Severe, args[1], TexlStrings.ErrStringExpected);
                 isValid = false;
             }
-            else if ((formatNode = args[1].AsStrLit()) != null)
+            else if (BinderUtils.TryGetConstantValue(checkTypesContext, args[1], out var formatArg))
             {
                 // Verify statically that the format string doesn't contain BOTH numeric and date/time
-                // format specifiers. If it does, that's an error accd to Excel and our spec.
-                var fmt = formatNode.Value;
+                // format specifiers. If it does, that's an error according to Excel and our spec.
 
                 // But firstly skip any locale-prefix
-                if (fmt.StartsWith("[$-"))
+                if (formatArg.StartsWith("[$-"))
                 {
-                    var end = fmt.IndexOf(']', 3);
+                    var end = formatArg.IndexOf(']', 3);
                     if (end > 0)
                     {
-                        fmt = fmt.Substring(end + 1);
+                        formatArg = formatArg.Substring(end + 1);
                     }
```

Comment:
No test case is reaching this portion.

Reply:
This hasn't changed, but since I'm touching this function I added a couple of cases that exercise this code as well.